Flint/src/main/java/flint/FlintRule.java

```diff
@@ -0,0 +1,22 @@
+package flint;
+
+import java.util.Collection;
+import java.io.RandomAccessFile;
+import java.io.IOException;
+
+import com.github.javaparser.ast.CompilationUnit;
+
+public abstract class FlintRule {
+
+  /**
+  * Runs this FlintRule instance and returns consequent LintFailures to the caller.
+  * 
+  * @param astRoot - the root of the abstract syntax tree of the input file
+  * @param inputFile - the input file to be run rules on
+  * @return Returns a collection of the consequent LintFailures of this FlintRule
+  * */
+  public static Collection<LintFailure> run(CompilationUnit astRoot, RandomAccessFile inputFile) {
+    System.out.println("static method can't be abstract.");
```

Comment:
I'm questioning whether it makes sense to require them to implement any method at all...
Since the user has full control over the configuration file, they should be able to implement it however they see fit.
I also don't think you can `@Override` static methods.
Maybe the better approach would be to implement our default methods so that there is a pattern to follow, but not place any restrictions on how to write rules?

Reply:
I believe having something like `run()` is important since we have to somehow find an entry point to start the linter that the user defined.

`run()` here I think should be the only thing that the driver would call, therefore I would say making it a non-static method would allow users to define their own rules in this case.

Reply:
The driver only calls methods in the `FlintConfiguration`.
The rules are out of scope for the driver, which leads me to think that it doesn't make sense to have an abstract rule class since the user is free to check for rules however they want.